### DISCUSSION TITLE
Is it possible to force https requests instead of http

### DISCUSSION THREAD

**Discussion by ghost** [ruffle-rs/ruffle#5774] [Q&A] [answered]
There are some flash games that send GET requests to a website using "http" instead of "https." 

It causes an error : "Mixed Content: The page at 'https://example.com' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://example.com'. This request has been blocked; the content must be served over HTTPS."

**Comment by danielhjacobs** [comment, marked as answer]
Yes, use the configuration option upgradeToHttps: https://github.com/ruffle-rs/ruffle/wiki/Using-Ruffle#configuration-options. It should be on-by-default on https websites already though.

**Reply by SporeProductions** [reply]
Does the polyfill version also have it on by default? I'm getting the same issue OP is reporting, using the version of Ruffle that's on your CDN.

**Reply by danielhjacobs** [reply]
Should be: https://github.com/ruffle-rs/ruffle/blob/87ce0f56b7a6e8c8aec533ded3716fddc2b319a3/web/src/navigator.rs#L31-L33 Applies when upgradeToHttps is set and the current page is hosted on https.

upgradeToHttps is default true: https://github.com/ruffle-rs/ruffle/blob/7cbba5bc8ff38a9a422664fd6b595af39b6bffe7/web/packages/core/src/load-options.ts#L182-L184

What page is having the issue?

**Reply by SporeProductions** [reply]
I was trying to figure out why this game wasn't proceeding and thought I saw it trying to call a banner ad from an http url and it being blocked. Today, after some cache-clearing and further examination I can't see the same error.

It's this game, "Flash3DX". It should proceed to display a start/play screen but instead stays stuck on the loading.
https://www.lilgames.com/games/puzzle/flash-3dx/

**Reply by n0samu** [reply]
@SporeProductions this is a bug in Ruffle. I've reported it here: https://github.com/ruffle-rs/ruffle/issues/7419
Hopefully the devs will be able to look into it!
